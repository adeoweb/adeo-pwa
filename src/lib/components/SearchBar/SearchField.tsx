import React, { ChangeEvent, Fragment, FunctionComponent } from 'react';
import { useTranslation } from 'react-i18next';

import { useSearchField } from 'src/peregrine/lib/talons/adeoweb/SearchBar/useSearchField';

type TSearchFieldProps = {
    onChange: (event: ChangeEvent<HTMLInputElement>) => void;
    onFocus: () => void;
    value: string;
    setFieldValue: (field: string, value: string) => void;
};

const SearchField: FunctionComponent<TSearchFieldProps> = ({
    onChange,
    onFocus,
    value,
    setFieldValue
}) => {
    const { t } = useTranslation();
    useSearchField({ setFieldValue });

    return (
        <Fragment>
            <input
                className="form-control"
                name="search_query"
                onFocus={onFocus}
                onChange={onChange}
                value={value}
                placeholder={t('Search...')}
            />
            <button className="search-button btn">
                <i className="icon-magnifier" />
            </button>
        </Fragment>
    );
};

export default SearchField;
