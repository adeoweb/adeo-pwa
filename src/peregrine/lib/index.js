// components
export { default as PeregrineContextProvider } from './PeregrineContextProvider';
