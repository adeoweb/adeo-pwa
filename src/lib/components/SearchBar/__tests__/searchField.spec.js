import { Form } from 'informed';

import React from 'react';

import { createTestInstance } from '@magento/peregrine';

import SearchField from '../SearchField';

const onChange = jest.fn();
const onFocus = jest.fn();
const setFieldValue = jest.fn();

test('renders correctly', () => {
    const instance = createTestInstance(
        <Form initialValues={{ search_query: '' }}>
            <SearchField
                setFieldValue={setFieldValue}
                location={{ pathname: '/' }}
                onChange={onChange}
                onFocus={onFocus}
                value="value"
            />
        </Form>
    );

    expect(instance.toJSON()).toMatchSnapshot();
});
