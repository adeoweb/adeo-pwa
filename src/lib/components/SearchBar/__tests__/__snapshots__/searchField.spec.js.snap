// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  onKeyDown={[Function]}
  onReset={[Function]}
  onSubmit={[Function]}
>
  <input
    className="form-control"
    name="search_query"
    onChange={[MockFunction]}
    onFocus={[MockFunction]}
    placeholder="Search..."
    value="value"
  />
  <button
    className="search-button btn"
  >
    <i
      className="icon-magnifier"
    />
  </button>
</form>
`;
